### PR TITLE
[Poetry][HOC] Save selected poem in progress

### PR DESCRIPTION
**Pull request by ajpal**
Heavily modeled after https://github.com/code-dot-org/code-dot-org/pull/25466

Saves the selected poem in the project sources so that it persists on page reload and share

Initial level load with default poem:
image

Share view with selected poem:
image

After remix with custom poem:
image

### CODE REVIEW

File: apps/src/p5lab/poetry/Poetry.js [hunk 1]
```diff
@@ -1,9 +1,21 @@
...
+import {POEMS} from './constants';
 
 export default class Poetry extends SpriteLab {
+  init(config) {
+    super.init(config);
+    if (config.level.selectedPoem) {
+      getStore().dispatch(setPoem(config.level.selectedPoem));
+    } else if (this.level.defaultPoem) {
+      getStore().dispatch(setPoem(POEMS[this.level.defaultPoem]));
+    }
```
Comment (maddiedierker) [inline apps/src/p5lab/poetry/Poetry.js]: this is subjective, but i think we can simplify this bit of code:

```javascript
const poem = config.level.selectedPoem || POEMS[this.level.defaultPoem];
if (poem) {
  getStore().dispatch(poem);
}
```

File: dashboard/public/pixelation/pixelation.js [hunk 2]
```diff
@@ -130,6 +130,10 @@ function initProjects() {
       getSelectedSong: function() {
         return false;
       },
+      setSelectedPoem: function() {},
```
Comment (maddiedierker) [inline dashboard/public/pixelation/pixelation.js:133]: i've run into this situation before (having to stub sourceHandler on pixelation) -- we should not have to do this 🙃 i've logged [STAR-1785](https://codedotorg.atlassian.net/browse/STAR-1785) to refactor this away

File: apps/src/p5lab/poetry/Poetry.js (hunk 1 above)
Comment (ajpal) [reply apps/src/p5lab/poetry/Poetry.js]: ah, yeah, I like that better! thanks!

File: dashboard/public/pixelation/pixelation.js (hunk 2 above)
Comment (ajpal) [reply dashboard/public/pixelation/pixelation.js:133]: thanks for logging that!

**Review by maddiedierker — APPROVED** [review]

**Pull request merged**